python/rmm/rmm/pylibrmm/memory_resource.pyx

```diff
@@ -203,6 +205,45 @@ cdef class CudaAsyncMemoryResource(DeviceMemoryResource):
         )
 
 
+cdef class CudaAsyncViewMemoryResource(DeviceMemoryResource):
+    """
+    Memory resource that uses ``cudaMallocAsync``/``cudaFreeAsync`` for
+    allocation/deallocation with an existing CUDA memory pool.
+
+    This resource uses an existing CUDA memory pool handle (such as the default pool)
+    instead of creating a new one. This is useful for integrating with existing GPU
+    applications that already use a CUDA memory pool, or customizing the flags
+    used by the memory pool.
+
+    Parameters
+    ----------
+    valid_pool_handle : cudaMemPool_t
+        Handle to a CUDA memory pool which will be used to serve allocation
+        requests.
+    """
+    def __cinit__(
+        self,
+        valid_pool_handle
+    ):
+        # Convert the valid_pool_handle to a cyruntime.cudaMemPool_t
+        cdef cyruntime.cudaMemPool_t c_valid_pool_handle
+        if isinstance(valid_pool_handle, (runtime.cudaMemPool_t, driver.CUmemoryPool)):
+            raw_pool_handle = int(valid_pool_handle)
+            c_valid_pool_handle = <cyruntime.cudaMemPool_t><uintptr_t>raw_pool_handle
+        else:
+            raw_pool_handle = int(runtime.cudaMemPool_t(valid_pool_handle))
```

Comment:
Is there any typing we can apply to the constructor argument that would prevent someone passing something bad in here (or does this constructor do that for hs?)